tests_new/integration_tests/modules/redshift_datasets/test_redshift_connection.py

```diff
@@ -0,0 +1,362 @@
+        client=client1, connection_uri=session_connection_serverless_admin.connectionUri
+    )
+    assert_that(len(response)).is_greater_than_or_equal_to(1)
+    assert_that(response).contains('public')
+
+
+def test_list_redshift_connection_schemas_unauthorized(client2, session_connection_serverless_admin):
+    assert_that(list_redshift_connection_schemas).raises(GqlError).when_called_with(
+        client=client2, connection_uri=session_connection_serverless_admin.connectionUri
+    ).contains(
+        'UnauthorizedOperation',
+        'GET_REDSHIFT_CONNECTION',
+        session_connection_serverless_admin.connectionUri,
+    )
+
+
+def test_list_redshift_schema_tables(client1, session_connection_serverless_admin):
+    response = list_redshift_schema_tables(
+        client=client1,
+        connection_uri=session_connection_serverless_admin.connectionUri,
+        schema='public',
+    )
+    assert_that(len(response)).is_greater_than_or_equal_to(1)
+    assert_that(response[0]).contains_key('name', 'type')
+
+
+def test_list_redshift_schema_tables_unauthorized(client2, session_connection_serverless_admin):
+    assert_that(list_redshift_schema_tables).raises(GqlError).when_called_with(
+        client=client2, connection_uri=session_connection_serverless_admin.connectionUri, schema='public'
+    ).contains(
+        'UnauthorizedOperation',
+        'GET_REDSHIFT_CONNECTION',
+        session_connection_serverless_admin.connectionUri,
+    )
+
+
+def test_list_redshift_connection_group_permissions(
+    client1, group1, session_connection_serverless_admin, session_connection_serverless_admin_group_with_permissions
+):
+    response = list_redshift_connection_group_permissions(
+        client=client1,
+        connection_uri=session_connection_serverless_admin.connectionUri,
+    )
+    assert_that(response.count).is_equal_to(2)
+    assert_that(response.nodes).extracting('groupUri').contains(
+        group1, session_connection_serverless_admin_group_with_permissions
+    )
+    response = list_redshift_connection_group_permissions(
+        client=client1,
+        connection_uri=session_connection_serverless_admin.connectionUri,
+        filter={'term': session_connection_serverless_admin_group_with_permissions},
+    )
+    assert_that(response.count).is_equal_to(1)
+    assert_that(response.nodes).extracting('groupUri').contains(
+        session_connection_serverless_admin_group_with_permissions
+    )
+
+
+def test_list_redshift_connection_group_permissions_unauthorized(client2, session_connection_serverless_admin):
+    assert_that(list_redshift_connection_group_permissions).raises(GqlError).when_called_with(
+        client=client2, connection_uri=session_connection_serverless_admin.connectionUri
+    ).contains(
+        'UnauthorizedOperation',
+        'MANAGE_REDSHIFT_CONNECTION_PERMISSIONS',
+        session_connection_serverless_admin.connectionUri,
+    )
+
+
+def test_list_redshift_connection_group_no_permissions(
```

Comment:
a little confused by this test ( `test_list_redshift_connection_group_no_permissions`) compared to (`test_list_redshift_connection_group_permissions`)

In `test_list_redshift_connection_group_no_permissions` and `test_list_redshift_connection_group_permissions` I think we run the same `list_redshift_connection_group_permissions` call but in one we expect 2 reponses including `group1, session_connection_serverless_admin_group_with_permissions` and in the other we expect it not to contain the same?

What am I missing here?

Reply:
- `test_list_redshift_connection_group_permissions` ---> it should return group1 because it has permissions to the connection
- `test_list_redshift_connection_group_no_permissions` ---> it should return all groups in the environment without permissions to the connection - it should not contain group1